src/invidious/routes/feeds.cr

```diff
@@ -128,6 +128,7 @@ module Invidious::Routes::Feeds
       watched = user.watched.reverse[(page - 1) * max_results, max_results]
     end
     watched ||= [] of String
+    watched = watched.map { |id| get_video(id) }
```

Comment:
Do mind that when you're calling `get_video`, you're asking invidious to get the entire watch page information from Youtube. Although these infos are cached, the chance of cache hit are thing, as the cache is regularly discarded to keep the video playback URLs "fresh" (these expire after 6h).

The biggest issue here is the potential privacy leak. On a very large instance, you likely hit the cache for some of these videos, so only a part of the watch history will be revealed, but on a small instance, you're pretty much guaranteed to send their entire watch history to Youtube whenever they load the page.

The second problem is performance. It works in reverse of the previous issue, meaning that this will impact large instances more. Even with the cache, imagine that if you have 200 users trying to see their 1000 videos-long watch history, that's 200 000 requests (well, actually 400 000) to Youtube! This will cripple down any medium or large sized instance.

Reply:
Performance is definitively something that should be improved (maybe paging would be a good idea to improve this and the privacy issues?, I would like your opinions on that :).

About the privacy leak: when the client loads the thumbnails, the watch history is also sent to YouTube (although in a more "indirect" way). Is that any different to loading the page when talking about privacy?